common/beerocks/bwl/common/ap_wlan_hal.h

```diff
@@ -261,6 +296,14 @@ class ap_wlan_hal : public virtual base_wlan_hal {
      */
     virtual bool read_acs_report() = 0;
 
+    /*!
+     * Read the supported channls from the hardware
+     * On successful completion the information can be retrieved 
+     *
+     * @return true on success or false on error.
+     */
+    virtual bool read_supported_channels() = 0;
```

Comment:
Why is this not just done on initialisation?

Reply:
What is not done in init? reading the supported channels? It can't be done on init because bwl needs to be attached first.

Reply:
> What is not done in init? reading the supported channels?

Yes, why is there a need to call this function explicitly, instead of just doing it.

> It can't be done on init because bwl needs to be attached first.

OK, then why is it not done on attach?

Anyway, I think the entire BWL should be looked at at a bit a higher level to redefine it as a clean API. So I created #477 for now, and the current situation is OK for now.